torch/csrc/autograd/engine.cpp

```diff
@@ -277,10 +302,22 @@ auto Engine::thread_main(
   // Why the test on graph_task->outstanding_tasks_?  See
   // Note [Reentrant backwards]
   while (!reentrant_thread || graph_task->outstanding_tasks_ > 0) {
-    NodeTask task = queue->pop();
+    std::unique_ptr<NodeTask> task{nullptr};
+    if (graph_task) {
+      // For the case where there're more re-entrant threads than outstanding
```

Comment:
I'm not sure to understand this comment.
These are not "re-entrant" threads but threads from the pool no?

Reply:
Re-entrants don't have to be in a different thread pool. The base behavior is that the current thread gets here.
So the case here is if some other thread in the multithread pool stole the work from this thread, this thread should not block because it needs to keep checking if his work got done by someone else?